assets/src/edit-story/utils/nativeCopyPasteExpected.js

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Checks if the native copy paste should proceed,
+ * either if text is selected or if there's focus on a text/number input.
+ *
+ * @return {boolean} If native handling is expected.
+ */
+function nativeCopyPasteExpected() {
+  const { activeElement } = document;
+  const { tagName, type, isContentEditable } = activeElement;
+  const isInput = () => {
+    return (
+      'textarea' === tagName.toLowerCase() ||
+      (('input' === tagName.toLowerCase() || 'true' === isContentEditable) &&
+        ('text' === type || 'number' === type))
+    );
+  };
+
+  if (isInput()) {
+    return true;
+  }
+
+  const selection = window.getSelection();
+  const range = selection.rangeCount ? selection.getRangeAt(0) : null;
+
+  return range && !range.collapsed;
```

Comment:
Perhaps return `Boolean(!range?.collapsed)`?

At minimum do the boolean cast, to match the jsdoc above. Right now it can return `null`.

Reply:
Yep, makes sense. `return Boolean(!range?.collapsed)` would actually give false results sometimes (e.g. it would give `true` if `range` is `null`), will cast it to Boolean though 👍